compiler-core/src/io/memory.rs

```diff
@@ -96,7 +115,49 @@ impl FileSystemWriter for InMemoryFileSystem {
         panic!("unimplemented") // TODO
     }
 
-    fn mkdir(&self, _: &Utf8Path) -> Result<(), Error> {
+    fn mkdir(&self, path: &Utf8Path) -> Result<(), Error> {
+        let mut parent_counter = 0;
+
+        // Traverse ancestors from parent to root
+        // Create each missing ancestor
+        for ancestor in path.ancestors() {
+            if ancestor == "" || ancestor.ends_with(".") {
+                continue;
+            }
+            if ancestor.ends_with("..") {
```

Comment:
What's this for? I thought we always used absolute paths.

Reply:
It was just because we're using camino which handles `..` and `.` differently, so I wanted to ensure correctness, but I guess this isn't respected elsewhere either, so I'll remove it